tests/TMS/CourseCreation/LinkHelperTest.php

```diff
@@ -0,0 +1,328 @@
+	}
+}
+
+
+/**
+ * @group needsInstalledILIAS
+ */
+class LinkHelperTest extends TestCase {
+	public static function setUpBeforeClass() {
+		require_once("./Services/User/classes/class.ilObjUser.php");
+		require_once("./Services/Language/classes/class.ilLanguage.php");
+
+		if(file_exists(
+				"./Customizing/global/plugins/Services/Cron/CronHook/CourseCreation/classes/class.ilCourseCreationPlugin.php"
+			)
+		) {
+			require_once("./Customizing/global/plugins/Services/Cron/CronHook/CourseCreation/classes/class.ilCourseCreationPlugin.php");
+		}
+	}
+
+	public function test__noOpenRequests() {
+		$usr = $this->getMockBuilder(ilObjUser::class)
+			->disableOriginalConstructor()
+			->getMock();
+
+		$link_helper = $this->getMockBuilder(LinkHelperMock::class)
+			->setMethods(array("getUsersDueRequests", "getUser", "getCourseCreationPlugin", "sendInfo"))
+			->getMock();
+
+		$link_helper->expects($this->never())
+			->method("sendInfo");
+
+		$link_helper->expects($this->once())
+			->method("getUsersDueRequests")
+			->will($this->returnValue(array()));
+
+		$link_helper->expects($this->once())
+			->method("getCourseCreationPlugin")
+			->will($this->returnValue(null));
+
+		$link_helper->expects($this->once())
+			->method("getUser")
+			->will($this->returnValue($usr));
+
+		$this->assertNull($link_helper->_maybeShowRequestInfo());
+	}
+
+	public function test_openRequests() {
+		$txt_message = "This is the user info";
+
+		$lng = $this->getMockBuilder(ilLanguage::class)
+			->setMethods(array("txt"))
+			->disableOriginalConstructor()
+			->getMock();
+
+		$lng->expects($this->once())
+			->method("txt")
+			->will($this->returnValue($txt_message));
+
+		$usr = $this->getMockBuilder(ilObjUser::class)
+			->disableOriginalConstructor()
+			->getMock();
+
+		$link_helper = $this->getMockBuilder(LinkHelperMock::class)
+			->setMethods(
+				array(
+					"getUsersDueRequests"
+					, "getUser"
+					, "getCourseCreationPlugin"
+					, "sendInfo"
+					, "getTrainingTitleByRequest"
+					, "getLng"
+				)
+			)
+			->getMock();
+
+		$request = $this->getMockBuilder(\ILIAS\TMS\CourseCreation\Request::class)
+			->disableOriginalConstructor()
+			->getMock();
+
+		$link_helper->expects($this->once())
+			->method("getUsersDueRequests")
+			->will($this->returnValue(array($request)));
+
+		$link_helper->expects($this->once())
+			->method("getCourseCreationPlugin")
+			->will($this->returnValue(null));
+
+		$link_helper->expects($this->once())
+			->method("getTrainingTitleByRequest");
+
+		$link_helper->expects($this->once())
+			->method("sendInfo")
+			->with($this->equalTo($txt_message));
+
+		$link_helper->expects($this->once())
+			->method("getLng")
+			->will($this->returnValue($lng));
+
+		$link_helper->expects($this->once())
+			->method("getUser")
+			->will($this->returnValue($usr));
+
+		$this->assertNull($link_helper->_maybeShowRequestInfo());
+	}
+
+	public function test_openRequestsWithPlugin() {
```

Comment:
Das hier testet bis in den ILIAS-Kern rein und wirft bei mir dabei Fehler. Die Tests failen also wieder. Nils hat beim letzten Teammeeting Strategien gezeigt um gegen ILIAS abzuschirmen. Die können hier ebenfalls genutzt werden.